kolibri/core/public/api.py

```diff
@@ -324,3 +327,35 @@ def create(self, request):
 
     def update(self, request, pk=None):
         return self.check_queue(request, pk=pk)
+
+
+class FacilitySearchUsernameViewSet(ReadOnlyValuesViewset):
+    filter_backends = (DjangoFilterBackend, filters.SearchFilter)
+    filter_fields = ("facility",)
+    search_fields = ("^username",)
+
+    values = ("id", "username")
+
+    def list(self, request, *args, **kwargs):
+        facility_id = request.query_params.get("facility", None)
+        try:
+            facility = Facility.objects.get(id=facility_id)
+        except (AttributeError, Facility.DoesNotExist):
+            # non existing facility
+            return Response({})
+
+        if facility.dataset.learner_can_login_with_no_password:
+            queryset = self.filter_queryset(self.get_queryset())
+            return Response(self.serialize(queryset))
+        else:
+            username = request.query_params.get("search", None)
+            queryset = self.get_queryset().filter(
+                facility=facility_id, username=username
+            )
+            response = {"username": username, "id": None} if queryset else {}
```

Comment:
I think we should return a list here rather than a single object to give a consistent return object regardless of the facility setting.

Reply:
fully agree, thanks!